port/zos390/omrvmem.c

```diff
@@ -1378,17 +1407,25 @@ omrvmem_find_valid_page_size(struct OMRPortLibrary *portLibrary, uintptr_t mode,
 #endif /* OMR_ENV_DATA64 */
 
 	/* If there is no default page size (or for executable pages), try 1MB pageable pages */
-	pageSizeFound = isLargePageSizeSupported(portLibrary, ONE_M, OMRPORT_VMEM_PAGE_FLAG_PAGEABLE);
-	if (TRUE == pageSizeFound) {
-		validPageSize = ONE_M;
-		SET_PAGE_TYPE(validPageFlags, OMRPORT_VMEM_PAGE_FLAG_PAGEABLE);
-	} else {
-		/* Use default page size, if everything else fails */
-		validPageSize = PPG_vmem_pageSize[0];
-		validPageFlags = PPG_vmem_pageFlags[0];
-	}
 
+	if (FALSE == isPageableAttempted) {
+		pageSizeFound = isLargePageSizeSupported(portLibrary, ONE_M, OMRPORT_VMEM_PAGE_FLAG_PAGEABLE);
+	
+		if (TRUE == pageSizeFound) {
+			validPageSize = ONE_M;
+			SET_PAGE_TYPE(validPageFlags, OMRPORT_VMEM_PAGE_FLAG_PAGEABLE);
+		} else {
+			/* Use default page size, if everything else fails */
+			validPageSize = PPG_vmem_pageSize[0];
+			validPageFlags = PPG_vmem_pageFlags[0];
+		}
+	}
 _end:
+	/* Reset pageFlags */
+	if (IS_VMEM_PAGE_FLAG_PAGEABLE_PREFERABLE(*pageFlags)) {
+		*pageFlags = OMRPORT_VMEM_PAGE_FLAG_NOT_USED;
+	}
+
```

Comment:
Please double check but I believe you don't need this any more. I claim there is no case `OMRPORT_VMEM_PAGE_FLAG_PAGEABLE_PREFERABLE` flag would be set at this point

Reply:
Sorry, I mean move this to the top of function right after flag has been cached in `validPageFlags`
Otherwise my comment does not have sense